augur/data/schema-multiplot.json

```diff
@@ -0,0 +1,126 @@
+                            "descriptions": "Default collection option where the value must be the `key` in one of the objects of the collections array. Optional -- if not provided, first collection option will be used",
+                            "type": "string"
+                        },
+                        "threshold": {
+                            "description": "Default options for the measurement threshold. Optional -- if not provided then threshold will not be included and threshold controls will be hidden",
+                            "type": "object",
+                            "additionalProperties": false,
+                            "required": ["show", "value"],
+                            "properties": {
+                                "show": {
+                                    "description": "Should the threshold line be displayed by default?",
+                                    "type": "boolean"
+                                },
+                                "value": {
+                                    "description": "The numeric value of the measurements threshold",
+                                    "type": "number"
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "collections": {
+            "description": "Collections of measurements of different types, conditions, adn/or experiments",
+            "type": "object",
+            "minProperties": 1,
+            "patternProperties": {
+                "^.*$": {
+                    "description": "A single collection of measurements",
+                    "type": "array",
+                    "minItems": 1,
+                    "items": {
+                        "description": "A single measurement for a sample and it's associated metadata. At least one of the metadata properties should be a group by option",
+                        "type": "object",
+                        "required": ["strain", "value"],
+                        "minProperties": 3,
+                        "properties": {
+                            "strain": {
+                                "description": "The name of the sample that matches the name of the sample within the tree",
+                                "type": "string"
+                            },
+                            "value": {
+                                "description": "The numeric value of the measurement",
+                                "type": "number"
+                            }
+                        },
+                        "patternProperties": {
+                            "^.*$": {
+                                "description": "Metadata associated with the measurement. Only metadata properties included in the groupings config will be included in the group by dropdown, but all metadata properties will be available as a filter",
```

Comment:
How should we order the values of a given grouping in the viz itself? I can think of a few ways, but not sure of the best:

1. _If_ the value appears in the tree (see another comment for more on this) then mirror the y-value ordering of the (rectangular) tree
2. Define this in the config, e.g.

```
"groupings": [
    {
        "key": "serum",
        "value": "Reference serum",
        "order": [“3C.2a2”, “3C.2a1b.2”, ...]
    },
```

3. Same way we order the legend currently: based on the number of measurements for each grouping

Reply:
(3) makes sense as the default, but maybe the user can override the default ordering with (2).

Reply:
Ordering by tree placement might be the most helpful default. The feedback we've received in the static antigenic distance versions of these plots is that the reference sera need to be grouped by clade and, preferably, ordered by those clades in a meaningful way. Ordering by tree placement would place closely related reference strains close to each other on the y-axis which would facilitate comparisons of similarly effective sera. 

Ordering by descending number of measurements could be helpful for titer data, since it would place frequently assayed references at the top of the plot, but there are often genetically similar references we want to compare to each other that have very different numbers of measurements (e.g., one reference that every CC has assayed and a bunch of candidates that only individual CCs have assayed with a handful of test viruses). This default would place those references at opposite ends of the ordering, making them harder to compare.

The second option was originally the most appealing to me, if we could precalculate the order by a summary statistic per reference or calculate that value on the fly. For example, if we could order references by ascending mean log2 titer distance across all test strains, we could find which serum best covers the selected test strains on average. Here, I was thinking of how Vega supports calculating new fields as a transform of the data and [ordering by an preexisting or calculated field](https://vega.github.io/vega-lite/docs/sort.html), allowing you to calculate the mean of a group, for example, and then sort by it. But, in practice, we usually know which clade(s) we need to consider reference sera from, so we're filtering to those clades and looking for the best coverage within those groups. Ordering by tree might be closer to what we want in that common scenario.